### PR TITLE
Deprecate and rename `respawn` option to `autoRespawn`

### LINKED ISSUE #3276: rename `respawn` option to `autoRespawn`
for clarity, let's deprecate `respawn` option in favor of more clear `autoRespawn`, also there is no way to disable `respawn`  disable ability in general so this might be confusing

from https://github.com/PrismarineJS/mineflayer/pull/3206#issuecomment-1737921758

### PR DESCRIPTION
**Pull request by zardoy**
fixes #3276

### CODE REVIEW

File: lib/loader.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ function createBot (options = {}) {
   options.loadInternalPlugins = options.loadInternalPlugins ?? true
   options.client = options.client ?? null
   options.brand = options.brand ?? 'vanilla'
-  options.respawn = options.respawn ?? true
+  options.autoRespawn = options.autoRespawn ?? options.autoRespawn ?? true
```
Comment (extremeheat) [inline lib/loader.js:73]: Incorrect logic

Also, the doc seems already pretty clear. Is there anyone that's been confused by this?

Comment (zardoy) [reply lib/loader.js]: oops, fixed

**Comment by rom1504** [comment]
why is it worth it to do such a breaking change for a minor renaming?

**Comment by zardoy** [comment]
> why is it worth it to do such a breaking change for a minor renaming?

where did you find the breaking change?

the pr is here because of https://github.com/PrismarineJS/mineflayer/issues/3276#issuecomment-1881884816, if for some reason you don't like it feel free to close it...

**Comment by rom1504** [comment]
Removing the option `options.respawn` is a breaking change

**Comment by zardoy** [comment]
> Removing the option options.respawn is a breaking change

But I didn't remove support for options.respawn, old option should still be supported. am I missing something?

**Comment by rom1504** [comment]
ah yeah I see, yeah then it's not breaking

not sure if it's really useful though, but maybe...

**Comment by zardoy** [comment]
Closing this so you can prioritize a lot of other more important prs

**Pull request closed without merging**